src/main/java/io/github/jav/exposerversdk/PushClient.java

```diff
@@ -38,20 +43,49 @@ public URL getBaseApiUrl() {
         return baseApiUrl;
     }
 
-    public CompletableFuture<List<ExpoPushTicket>> sendPushNotificationsAsync(List<Z> messages) {
+    public CompletableFuture<ExpoPushResponse> sendPushNotificationsAsync(List<Z> messages) {
         try {
+            
+            
             return _postNotificationAsync(new URL(baseApiUrl + "/push/send"), messages)
                     .thenApply((String jsonString) -> {
                         try {
                             ObjectMapper mapper = new ObjectMapper();
-                            JsonNode dataNode = mapper.readTree(jsonString).get("data");
-                            List<ExpoPushTicket> retList = new ArrayList<>();
-                            for (JsonNode node : dataNode) {
-                                retList.add(mapper.convertValue(node, ExpoPushTicket.class));
+                            ExpoPushResponse response = mapper.readValue(jsonString,  ExpoPushResponse.class);
```

Comment:
Would you mind adding one or more unit-tests for this section?

Reply:
I will add tests and documentation if you are okay with the design philosophy. This code has been used in my production environment with no errors so far. I'm going on vacation will be back next Tuesday, can work on it then if you agree

Reply:
I've given it more thought. :)

I _think_ that most users won't use the 'data' field.
Which leads me to suggest the following setup (for which I'd like to hear your opinion).

`ExpoPushMessage.data` changes type to `Map<String, Object>`, which aligns with [the API spec](https://docs.expo.io/push-notifications/sending-notifications/#message-request-format)
I also _think_ that jackson will be able to figure out how to serialize/deserialize `Object` by it self in most cases.

If a user (a power user) wants to change the type of `data`, they'll need to extend the class  `ExpoPushMessage`.

Below is an example (which won't run out of the box, it assumes that I get rid of most `.get` & `.set` methods, which right now are unnecessary anyhow. I'm refactoring this in a different branch).

```
package io.github.jav.exposerversdk.example;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import io.github.jav.exposerversdk.ExpoPushMessage;

import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"_debug"})
public class ExampleExpoPushMessageWithOverriddenDataType extends ExpoPushMessage {
    public Map<String, Integer> data;

    ExampleExpoPushMessageWithOverriddenDataType() {
        super();
        data = new HashMap<>();
    }
}
`` 

This example works fine with my unittests :)


I'm assuming that:
Most users don't use the data field.
That most users find Generics (and inheritance) hard to reason about.

That Jackson will be able to serialize/de-serialize most objects with class `Object`



BTW, if I haven't already thanked you for the initiative & contribution.
Thank you!

Enjoy your vacation!